File: Swift/lambda - closure/3/main.swift
//
//  main.swift
//  s
//
//  Created by Fay on 2017/6/2.
//  Copyright © 2017年 Fay. All rights reserved.
//

import Foundation

func say(_ reply: (String) -> String) {
    print(reply("Hello World!"))
}

say { (content) in
    print(content)
    return "Hello Swift!"
}

File: Rust/lambda - closure/3/main.rs
fn main() {
    say(&|end| {
        println!("Hello Rust!");
        end("^_^".to_string());
    })
}

fn say(reply: &Fn(&Fn(String))) {
    println!("Hello World!");
    reply(&|content| {
        println!("{}", content);
    })
}
